tests/communication_tests.cpp

```diff
@@ -0,0 +1,35 @@
+#include <iostream>
+#include <thread>
+#include <chrono>
+#include <unistd.h>
+#include <tuple>
+#include "doctest.h"
+#include "pseudoterminal.h"
+#include "LinSerial.h"
```

Comment:
how many of these headers are actually needed?

Reply:
I did a general cleanup of the headers now